### PR TITLE
😳 Add New Processor

### PR DESCRIPTION
**Pull request by dathudeptrai**
This PR aim to add new base processor class to easy inherit, custom.

- [x] Add new base processor class. 

- [x] Re-new ljspeech/kss/baker/libritts Processor. 

- [x] Add option to save/load from pretrained processor json file. 

- [x] Add AutoProcessor class. 

- [x] Solve the problem about `eos` token ambiguous.

### CODE REVIEW

File: tensorflow_tts/processor/ljspeech.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 from tensorflow_tts.utils import cleaners
 from tensorflow_tts.processor import BaseProcessor
 
+from dataclasses import dataclass
+
 valid_symbols = [
```
Comment (machineko) [inline tensorflow_tts/processor/ljspeech.py:16]: Let's load it from file maybe :D 
```python3
with open("prepro/SOMEPATH", "r") as f:
    valid_symbols = f.readlines()
```

Or save all symbols and mapper to json file and load it using processor class 📦 also uploading it to repo

File: tensorflow_tts/processor/base_processor.py [hunk 2]
```diff
@@ -23,17 +23,17 @@ class BaseProcessor(abc.ABC):
...
     # extras
     items: List[List[str]] = field(default_factory=list)  # text, wav_path, speaker_name
     symbol_to_id: Dict[str, int] = field(default_factory=dict)
     id_to_symbol: Dict[int, str] = field(default_factory=dict)
 
     def __post_init__(self):
 
-        if self.load_mapper:
-            self._load_mapper()
+        if self.loaded_mapper_path is not None:
```
[the remark this answers is not in the document]
Comment (dathudeptrai) [reply tensorflow_tts/processor/base_processor.py:35]: i know :)) just my habit :))

File: tensorflow_tts/processor/base_processor.py [hunk 3]
```diff
@@ -178,7 +178,13 @@ def _load_mapper(self, loaded_path: str = None):
         self.symbol_to_id = data["symbol_to_id"]
         self.id_to_symbol = {int(k): v for k, v in data["id_to_symbol"].items()}
 
-    def _save_mapper(self, saved_path: str = None):
+        # other keys
```
Comment (dathudeptrai) [inline tensorflow_tts/processor/base_processor.py:181]: @machineko let take a look if it make sense :)).

File: tensorflow_tts/processor/base_processor.py [hunk 4]
```diff
@@ -193,4 +199,6 @@ def _save_mapper(self, saved_path: str = None):
                 "id_to_symbol": self.id_to_symbol,
                 "speakers_map": self.speakers_map,
             }
+            if extra_attrs_to_save:
```
Comment (dathudeptrai) [inline tensorflow_tts/processor/base_processor.py:202]: @machineko let take a look if it make sense :)).

File: tensorflow_tts/processor/base_processor.py (hunk 3 above)
Comment (machineko) [reply tensorflow_tts/processor/base_processor.py:181]: LGFM

File: tensorflow_tts/processor/base_processor.py (hunk 4 above)
Comment (machineko) [reply tensorflow_tts/processor/base_processor.py:202]: LGFM

**Comment by azraelkuan** [comment]
i have test using cn dataset, works well

**Pull request merged**